lib/Sema/DerivedConformanceComparable.cpp

```diff
@@ -100,13 +93,11 @@ deriveBodyComparable_enum_noAssociatedValues_lt(AbstractFunctionDecl *ltDecl,
                                          /*Implicit*/ true,
                                          TupleType::get(abTupleElts, C));
 
-  auto *cmpExpr = new (C) BinaryExpr(
-      cmpFuncExpr, abTuple, /*implicit*/ true,
-      fnType->castTo<FunctionType>()->getResult());
+  auto *cmpExpr = new (C) BinaryExpr(cmpFuncExpr, abTuple, /*Implicit*/ true);
   statements.push_back(new (C) ReturnStmt(SourceLoc(), cmpExpr));
 
   BraceStmt *body = BraceStmt::create(C, SourceLoc(), statements, SourceLoc());
-  return { body, /*isTypeChecked=*/true };
+  return {body, /*isTypeChecked=*/false};
```

Comment:
Changed `isTypechecked` to `false` because this synthesizer is also used from `deriveBodyComparable_enum_hasAssociatedValues_lt()` below, which is non-typechecked.